react/config-overrides.js.template

```
@@ -18,8 +18,14 @@ module.exports = function override(config, env) {
       ignoreFile: '.sentrycliignore',
       ignore: ['webpack.config.js'],
       configFile: 'sentry.properties',
-      reactComponentAnnotation: {enabled:true},
+    })
+  );
+  config.plugins.push(
+    codecovWebpackPlugin({
+      enableBundleAnalysis: process.env.CODECOV_TOKEN !== undefined,
```

ideally we'd have some other condition on the environment here (e.g. `process.env.CI` `process.env.REACT_APP_ENVIRONMENT`) otherwise this can mask configuration issues (CODECOV_TOKEN not set by mistake)

maybe something like this so you can test it locally if you want to by setting the token?
```
process.env.CI !== undefined || process.env.CODECOV_TOKEN !== undefined
```

---

Shouldn't it be "&&" instead of "||"? Because otherwise we would enable BA if CI has a value regardless of the CODECOV_TOKEN.
`process.env.CI !== undefined && process.env.CODECOV_TOKEN !== undefined`

---

> otherwise we would enable BA if CI has a value regardless of the CODECOV_TOKEN.
yes that way it won't silently skip the step leaving us scratching our heads and debugging why it's not working.
it should fail in CI if CODECOV_TOKEN is not set, ideally in a way that's obvious how to fix it.